ecommerce/utils.py

```diff
@@ -79,7 +82,7 @@ def make_checkout_url(*, product_id=None, code=None, run_tag=None):
     if product_id is None and code is None:
         return base_checkout_url
 
-    query_params = {}
+    query_params = {"is_voucher_applied": is_voucher_applied}
```

Comment:
We can write a test if the voucher application boolean is passed the generated checkout URL contains that.

Reply:
There were no existing tests for this util. I was about to add one for all the query parameters using parametrization, but it seems useless because it will contain all of the logic from the function. Do you think we need to add it?

```
@pytest.mark.parametrize(
    ("product_id", "coupon_code", "run_tag", "is_voucher_applied"),
    [
        (
            1,
            "test_coupon_code",
            "R1",
            True,
        ),
        (
            None,
            "test_coupon_code",
            "R1",
            True,
        ),
        (
            1,
            None,
            "R1",
            True,
        ),
        (
            None,
            None,
            "R1",
            True,
        ),
        (
            1,
            "test_coupon_code",
            None,
            True,
        ),
        (
            None,
            "test_coupon_code",
            None,
            True,
        ),
        (
            1,
            None,
            None,
            True,
        ),
        (
            None,
            None,
            None,
            True,
        ),
        (
            1,
            "test_coupon_code",
            "R1",
            False,
        ),
        (
            None,
            "test_coupon_code",
            "R1",
            False,
        ),
        (
            1,
            None,
            "R1",
            False,
        ),
        (
            None,
            None,
            "R1",
            False,
        ),
        (
            1,
            "test_coupon_code",
            None,
            False,
        ),
        (
            None,
            "test_coupon_code",
            None,
            False,
        ),
        (
            1,
            None,
            None,
            False,
        ),
        (
            None,
            None,
            None,
            False,
        ),
    ],
)
def test_make_checkout_url(
    settings, product_id, coupon_code, run_tag, is_voucher_applied
):
    """Test make_checkout_url"""
    base_checkout_url = urljoin(settings.SITE_BASE_URL, reverse("checkout-page"))
    if product_id is None and coupon_code is None:
        return base_checkout_url

    query_params = {"is_voucher_applied": is_voucher_applied}
    if product_id is not None:
        query_params["product"] = (
            product_id
            if run_tag is None
            else f"{product_id}{ENROLLABLE_ITEM_ID_SEPARATOR}{run_tag}"
        )
    if coupon_code is not None:
        query_params["code"] = coupon_code

    assert (
        make_checkout_url(
            product_id=product_id,
            code=coupon_code,
            run_tag=run_tag,
            is_voucher_applied=is_voucher_applied,
        )
        == f"{base_checkout_url}?{urlencode(query_params)}"
    )
```

Reply:
What we need to make sure that the actual function is working, What we can do is in the params we add another one a hardcoded URL that we will be expecting it to be. If we can't add it to params we can expect/assert a hard coded URL from the make_checkout_url